### ISSUE TITLE
`authentication signature from unknown issuer` on `tofu init` using Cloudflare provider v2.27.0

### ISSUE THREAD

**Issue by danxmoran** [opentofu/registry#680]
### Thank you for reporting an issue.

I'm migrating from `terraform` to `tofu`. I have a module that uses `cloudflare/cloudflare` - when I run `tofu init` in it, I get:
```
Initializing provider plugins...
- Finding cloudflare/cloudflare versions matching "~> 2.21"...
- Installing cloudflare/cloudflare v2.27.0...
╷
│ Error: Failed to install provider
│
│ Error while installing cloudflare/cloudflare v2.27.0: authentication signature from unknown issuer
```

### OpenTofu Version

```shell
OpenTofu v1.6.2
on darwin_arm64

# NOTE: Using v1.6.2 temporarily as part of the recommended TF->Tofu migration process
```

### OpenTofu Configuration Files

provider.tf:
```hcl
terraform {
  required_version = "1.6.2"
  required_providers {
    cloudflare = {
      version = "~> 2.21"
      source = "cloudflare/cloudflare"
    }
  }
}
```

**Comment by cam72cam** [comment]
This is an issue where cloudflare has only provided the "latest" provider key and not the legacy keys.  We need to reach out to them to provide all of the keys they have used to sign providers in the past.

Until they provide their legacy keys, you will need to upgrade to a version later than 3.9.1

**Comment by cam72cam** [comment]
I've opened up https://github.com/cloudflare/terraform-provider-cloudflare/issues/3366 to track this with cloudflare

**Comment by jacobbednarz** [comment]
the legacy keys have been located and have been added via #684.

as a side note here @danxmoran, i would strongly recommend upgrading from 2.x as it is now 2 major versions behind and will not be receiving support for bugs or features.

**Comment by ghost** [comment]
Thank you @jacobbednarz , I've merged #684, this issue should be resolved within an hour. Closing this issue, please let us know if it is still a problem.

**Comment by danxmoran** [comment]
Thank you both!!

@jacobbednarz agreed on upgrading the provider - I've added it to my upgrade TODO list 😅
